README to DESCRIPTION, new README and CONTRIBUTING (#19)
G1ANT-Robot/G1ANT.Manual: CONTRIBUTING.md

# Contributing

> First of all, kudos to everyone, who wants to share their ideas, comments, complaints, flames and other input with the mighty Robot! The Robot appreciates any kind of contribution, which helps him to get rid of humans, er… to take the burden off of them.

Reviewer: First of all, we invite you to....

Author: ```suggestion
First of all, we invite you to contribute to this documentation and kudos to everyone, who wants to share their ideas, comments, complaints, flames and other input with the mighty Robot! The Robot appreciates any kind of contribution, which helps him to get rid of humans, er… to take the burden off of them.
```

When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change. The Robot will severely punish those who don’t comply.

Please note the G1ANT.Robot has a code of conduct, so please follow it in all your interactions with the project.

## Opening Issues

1. Use a standard GitHub issue tool to send a message to the Robot and make him aware of an issue.
2. Remember to attach screenshots, logs or other files that would help the Robot pinpoint the issue and/or reproduce it.

> 3. If the Robot admits you are right, which is possible, but highly unlikely, you will be granted a right to make a pull request. Or the Robot will make the necessary changes himself, since he is inerrant.

Reviewer 2: highly unlikely => highly unlike

It's a bit demotivating, isn't it? "If the Robot admits you're right, which you're probably not"... I understand that it is not serious, but still a bit awkward, especially because previous points are serious.

Author: you're right :smile: 
Ofc, I can change it if it's too much of a joke

Reviewer 2: @ChrisPrusik ?

Reviewer: @lfronczyk  I love you because of your comment!

Author: Alright then, I'll change it

Author: ```suggestion
3. If the Robot admits you are right, which is possible, you will be granted a right to make a pull request. Or the Robot will make the necessary changes himself, since he is inerrant.
```

## Pull Request Process

1. Ensure your request is well described, so that the Robot understands what you want to do. Otherwise the Robot will issue the 400 Bad Request error.

> 2. Be patient. The Robot will process your request eventually. But sooner than you think.

Reviewer: Remember about
- README.md file on each directory (what is README.md? one sentence)
- DESCRIPTION.md file if you wish (what is DESCRIPTION.md? one sentence)
- Where is the right place of screenshots, logs or other files?

Author: README.md is only one for the whole manual, not for each directory. Or I don't understand the purpose of it in each  directory.
DESCRIPTION and -assets are described in README. Shall I copy it here as well?

> 

Author: ```suggestion

## Documentation Build Process

1. G1ANT.Manual repository is the place for all documentation content that is not a description of individual G1ANT.Language and G1ANT.Addons components such as commands, variables, structures etc. If you want to write something on the Robot's user interface, real case scenarios, tips and tricks and so on, use the G1ANT.Manual repository.
2. Tutorials are stored in G1ANT.Tutorials repository. Use it when you want to add courses and/or short lessons for users showing practical use of G1ANT.Language.
3. All content related to G1ANT.Addons, including G1ANT.Language commands descriptions, is in the G1ANT.Addons repository. This repository relies on developers, who create addons and accompanying documentation.
4. Our sync tool gathers together all Markdown files from Manual, Tutorials and Addons repositories and creates content to be displayed on G1ANT.Manual website.

```

### Scope

This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.

### Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at support[at]g1ant[dot]com. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.

Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership.

### Attribution

> This Code of Conduct is adapted from the [Contributor Covenant](http://contributor-covenant.org/), version 1.4, available at [http://contributor-covenant.org/version/1/4](http://contributor-covenant.org/version/1/4/)

Reviewer: ### Build process

How our documentation is creating (step by step) and where is the destination version?

Reviewer 2: @ChrisPrusik are you asking about the documentation for our internal process of the Robot build?

Author: It's about how we build the manual for the website. I included the info on how it's done.

Please note the G1ANT.Robot has a code of conduct, so please follow it in all your interactions with the project.

## Opening Issues

1. Use a standard GitHub issue tool to send a message to the Robot and make him aware of an issue.
2. Remember to attach screenshots, logs or other files that would help the Robot pinpoint the issue and/or reproduce it.
3. If the Robot admits you are right, which is possible, you will be granted a right to make a pull request. Or the Robot will make the necessary changes himself, since he is inerrant.

## Pull Request Process

1. Ensure your request is well described, so that the Robot understands what you want to do. Otherwise the Robot will issue the 400 Bad Request error.
2. Be patient. The Robot will process your request eventually. But sooner than you think.

## Documentation Build Process

1. G1ANT.Manual repository is the place for all documentation content that is not a description of individual G1ANT.Language and G1ANT.Addons components such as commands, variables, structures etc. If you want to write something on the Robot's user interface, real case scenarios, tips and tricks and so on, use the G1ANT.Manual repository.

> 2. Tutorials are stored in G1ANT.Tutorials repository. Use it when you want to add courses and/or short lessons for users showing practical use of G1ANT.Language.

Author: ```suggestion
2. Tutorials are stored in the [G1ANT.Tutorials](https://github.com/G1ANT-Robot/G1ANT.Tutorials) repository. Use it when you want to add courses and/or short lessons for users showing practical use of G1ANT.Language. These materials make up the [G1ANT.Learning](http://learning.g1ant.com) website, which is also available from within the G1ANT.Robot interface. Tutorials are also a part of [G1ANT.Manual](https://manual.g1ant.com) website.
```

> 3. All content related to G1ANT.Addons, including G1ANT.Language commands descriptions, is in the G1ANT.Addons repository. This repository relies on developers, who create addons and accompanying documentation.

Author: ```suggestion
3. All content related to G1ANT.Addons, including G1ANT.Language commands descriptions, is in the [G1ANT.Addon](https://github.com/G1ANT-Robot/G1ANT.Addon) repository. This repository relies on developers, who create addons and accompanying documentation.
```

> 4. Our sync tool gathers together all Markdown files from Manual, Tutorials and Addons repositories and creates content to be displayed on G1ANT.Manual website.

Author: ```suggestion
4. Our sync tool gathers together all Markdown files from Manual, Tutorials and Addon repositories and creates content to be displayed on the [G1ANT.Manual](https://manual.g1ant.com) website.
```

- Other conduct which could reasonably be considered inappropriate in a professional setting

### Our Responsibilities

Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.

Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

### Scope

This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.

### Enforcement

> Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at support[at]g1ant[dot]com. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.

Reviewer 3: why does our mail look like this "support[at]g1ant[dot]com" and not simply support@g1ant.com?

Reviewer: Add urls: https://manual.g1ant.com and https://learning.g1ant.com and to GitHub repositories

Author: ```suggestion
Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at support@g1ant.com. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
